Cross Compile   rsyslog-8.2512.0 is broken for aarch64-linux

### 📝 Description / What's on your mind:

I want to compile from x86_64-unknown-linux-gnu to aarch64-unknown-linux-gnu.
I also want libgcrypt enabled, but when enable this flag the build brakes. 
_Note that if i disable libgcrypt the build works and rsyslog works as expected._


When building from source it fails during configuration
```
       > checking for pkg-config... no
       > checking for libgcrypt-config... no
       > configure: error: in '/build/rsyslog-8.2512.0':
       > configure: error: libgcrypt-config not found in PATH and pkg-config is not available. libgcrypt is missing.
       > See 'config.log' for more details
```

**"> checking for pkg-config... no"**
This part suprised me becouse $PKG_CONFIG and PKG_CONFIG_PATH are set:
```
$ echo $PKG_CONFIG
aarch64-unknown-linux-gnu-pkg-config

$ aarch64-unknown-linux-gnu-pkg-config --version
0.29.2

$ pkg-config   # native pkg-configis on path
bash: pkg-config: command not found
```

### All flags
```
  "--enable-libgcrypt" # <-THIS couses my build to fail during configuration
  "--build=x86_64-unknown-linux-gnu"
  "--host=aarch64-unknown-linux-gnu"
  "--sysconfdir=/etc"
  "--localstatedir=/var"
  "--with-systemdsystemunitdir=\${out}/etc/systemd/system"
  "--enable-largefile"
  "--enable-regexp"
  "--enable-gssapi-krb5"
  "--enable-klog"
  "--enable-kmsg"
  "--enable-imjournal"
  "--enable-inet"
  "--enable-jemalloc"
  "--enable-unlimited-select"
  "--enable-clickhouse"
  "--disable-debug"
  "--disable-debug-symbols"
  "--enable-debugless"
  "--disable-valgrind"
  "--disable-diagtools"
  "--enable-fmhttp"
  "--enable-usertools"
  "--enable-mysql"
  "--enable-pgsql"
  "--enable-libdbi"
  "--enable-snmp"
  "--enable-uuid"
  "--enable-elasticsearch"
  "--enable-gnutls"
  "--enable-rsyslogrt"
  "--enable-rsyslogd"
  "--enable-mail"
  "--enable-mmnormalize"
  "--enable-mmdblookup"
  "--enable-mmjsonparse"
  "--enable-mmaudit"
  "--enable-mmanon"
  "--enable-mmutf8fix"
  "--enable-mmcount"
  "--enable-mmsequence"
  "--enable-mmfields"
  "--enable-mmpstrucdata"
  "--enable-mmrfc5424addhmac"
  "--enable-relp"
  "--disable-ksi-ls12"
  "--enable-liblogging-stdlog"
  "--enable-rfc3195"
  "--enable-imfile"
  "--disable-imsolaris"
  "--enable-imptcp"
  "--enable-impstats"
  "--enable-omprog"
  "--enable-omudpspoof"
  "--enable-omstdout"
  "--enable-omjournal"
  "--enable-pmlastmsg"
  "--enable-pmcisconames"
  "--enable-pmciscoios"
  "--enable-pmaixforwardedfrom"
  "--enable-pmsnare"
  "--enable-omruleset"
  "--enable-omuxsock"
  "--enable-mmsnmptrapd"
  "--disable-omhdfs"
  "--enable-omkafka"
  "--enable-ommongodb"
  "--enable-imczmq"
  "--enable-omczmq"
  "--enable-omrabbitmq"
  "--enable-omhiredis"
  "--enable-omhttp"
  "--enable-generate-man-pages"
```

- **Rsyslog Version:rsyslog-8.2512.0** 
- **buildPlatform: x86_64-unknown-linux-gnu** 
- **hostPlatform: aarch64-unknown-linux-gnu**

as for my build [script ](https://github.com/GlennWSo/cross-rsylog) Im using nix and invoke it like so 




```bash
cross-rsylog on  main took 2s
❯ nix build  .#cross_rsyslog_working # without Gcrypt
Success

cross-rsylog on  main
❯ nix build  .#cross_rsyslog_2512
error: Cannot build '/nix/store/8hgmqva9wgzsp8rc5aflr2wxnhaxywri-rsyslog-aarch64-unknown-linux-gnu-8.2512.0.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/x6m1dslil0259srxk5rmaixkc1d0cfl0-rsyslog-aarch64-unknown-linux-gnu-8.2512.0
       Last 25 log lines:
       > checking for mysqlclient... yes
       > checking if we have mysql_library_init... checking for mysql_init in -lmysqlclient... yes
       > checking for libpq... yes
       > checking for PGsslInUse in -lpq... no
       > checking for dbi/dbi.h... yes
       > checking for dbi_initialize in -ldbi... yes
       > checking for dbi_initialize_r in -ldbi... yes
       > checking for dbi_conn_transaction_begin in -ldbi... yes
       > checking for net-snmp/net-snmp-config.h... yes
       > checking for snmp_timeout in -lnetsnmp... yes
       > checking for uuid... yes
       > checking for libcurl... yes
       > checking for cos in -lm... yes
       > checking for libcurl... yes
       > checking for cos in -lm... (cached) yes
       > checking for libcurl... yes
       > checking for cos in -lm... (cached) yes
       > checking for gnutls >= 1.4.0... yes
       > checking for gnutls_certificate_set_retrieve_function... yes
       > checking for gnutls_certificate_type_set_priority... no
       > checking for pkg-config... no
       > checking for libgcrypt-config... no
       > configure: error: in '/build/rsyslog-8.2512.0':
       > configure: error: libgcrypt-config not found in PATH and pkg-config is not available. libgcrypt is missing.
       > See 'config.log' for more details
       For full logs, run:
         nix log /nix/store/8hgmqva9wgzsp8rc5aflr2wxnhaxywri-rsyslog-aarch64-unknown-linux-gnu-8.2512.0.drv
```

`configure: error: libgcrypt-config not found in PATH and pkg-config is not available. libgcrypt is missing.`
this error msg leads me towards that my pkg-config is not deteced. But how can that be?

--- a/configure.ac
+++ b/configure.ac
@@ -1443,38 +1443,23 @@ AC_ARG_ENABLE(libgcrypt,
         [enable_libgcrypt=yes]
 )
 if test "x$enable_libgcrypt" = "xyes"; then
-        # Check for the modern pkg-config method first
-        AC_CHECK_PROG([HAVE_PKG_CONFIG], [pkg-config], [yes], [no])
-        if test "x${HAVE_PKG_CONFIG}" = "xno"; then
-                # Fallback to the outdated libgcrypt-config approach
-                AC_PATH_PROG([LIBGCRYPT_CONFIG],[libgcrypt-config],[no])
-                if test "x${LIBGCRYPT_CONFIG}" = "xno"; then
-                        AC_MSG_FAILURE([libgcrypt-config not found in PATH and pkg-config is not available. libgcrypt is missing.])
-                fi
-                AC_CHECK_LIB(
-                        [gcrypt],
-                        [gcry_cipher_open],
-                        [LIBGCRYPT_CFLAGS="`${LIBGCRYPT_CONFIG} --cflags`"
-                        LIBGCRYPT_LIBS="`${LIBGCRYPT_CONFIG} --libs`"
-                        ],
-                        [AC_MSG_FAILURE([libgcrypt is missing])],
-                        [`${LIBGCRYPT_CONFIG} --libs --cflags`]
-                )
-        else
-                # Use the preferred pkg-config approach
-                AC_CHECK_LIB(
-                        [gcrypt],
-                        [gcry_cipher_open],
-                        [LIBGCRYPT_CFLAGS="`pkg-config --cflags libgcrypt gpg-error`"
-                        LIBGCRYPT_LIBS="`pkg-config --libs libgcrypt gpg-error`"
-                        ],
-                        [AC_MSG_FAILURE([libgcrypt is missing])],
-                        [`pkg-config --libs --cflags libgcrypt gpg-error`]
-                )
-		if test "x${LIBGCRYPT_CFLAGS}" = "x" && test "x${LIBGCRYPT_LIBS}" = "x";then
-			AC_PATH_PROG([LIBGCRYPT_CONFIG],[libgcrypt-config],[no])
+        PKG_CHECK_MODULES(
+                [LIBGCRYPT],
+                [libgcrypt gpg-error],
+                [
+                        AC_CHECK_LIB(
+                                [gcrypt],
+                                [gcry_cipher_open],
+                                [],
+                                [AC_MSG_FAILURE([libgcrypt is missing])],
+                                [${LIBGCRYPT_LIBS}]
+                        )
+                ],
+                [
+                        # Fallback to the outdated libgcrypt-config approach
+                        AC_PATH_PROG([LIBGCRYPT_CONFIG],[libgcrypt-config],[no])
                         if test "x${LIBGCRYPT_CONFIG}" = "xno"; then
-                                AC_MSG_FAILURE([libgcrypt-config not found in PATH and pkgconfig file of libgcrypt is not available. libgcrypt is missing.])
+                                AC_MSG_FAILURE([libgcrypt-config not found in PATH and pkg-config is unavailable or could not find libgcrypt/gpg-error. libgcrypt is missing.])
                         fi
                         AC_CHECK_LIB(
                                 [gcrypt],
@@ -1485,8 +1470,8 @@ if test "x$enable_libgcrypt" = "xyes"; then
                                 [AC_MSG_FAILURE([libgcrypt is missing])],
                                 [`${LIBGCRYPT_CONFIG} --libs --cflags`]
                         )
-		fi
-        fi
+                ]
+        )
         AC_DEFINE([ENABLE_LIBGCRYPT], [1], [Indicator that LIBGCRYPT is present])
 fi
 AM_CONDITIONAL(ENABLE_LIBGCRYPT, test x$enable_libgcrypt = xyes)

